### PR TITLE
Updates Header formatting

### LINKED ISSUE #3260: Notifications: Apply Header Ranges
Let's apply the ranges available in the `header` field, so that the context in which a comment is made is easier to read.

Web Mockup below.  The username label will now include the source blog (`on [Blog Name]`).

screen shot 2015-02-13 at 10 26 40 am

### PR DESCRIPTION
**Pull request by jleandroperez**
Closes #3260

/cc @diegoreymendez (Thanks in advance Diego!)

### CODE REVIEW

File: WordPress/Classes/Extensions/Notifications/NotificationBlock+Interface.swift [hunk 1]
```diff
@@ -3,53 +3,72 @@ import Foundation
...
 extension NotificationBlock
 {
-    public func subjectAttributedText() -> NSAttributedString {
-        let richSubjectCacheKey = "richSubject"
-        
-        if let cachedSubject = cacheValueForKey(richSubjectCacheKey) as? NSAttributedString {
-            return cachedSubject
+    // MARK: - Text Formatting Helpers
+    //
+    public func attributedSubjectText() -> NSAttributedString {
```
Comment (diegoreymendez) [inline WordPress/Classes/Extensions/Notifications/NotificationBlock+Interface.swift]: Hey @jleandroperez, I'm still reviewing this, but one thing that could make things a bit easier is to document the class and its functions - not to explain what they're doing which should be figured out by the reviewer, but rather what they're supposed to do (ie: their purpose).

Comment (jleandroperez) [inline WordPress/Classes/Extensions/Notifications/NotificationBlock+Interface.swift]: Will do, thank you Diego!

**Comment by jleandroperez** [comment]
@diegoreymendez Documentation added, thank you!

File: WordPress/Classes/Extensions/Notifications/NotificationBlock+Interface.swift [hunk 2]
```diff
@@ -1,55 +1,120 @@
...
-            return cachedSubject
+    // MARK: - Public Methods
+    //
+    
+    /**
+    *	@brief      Formats a NotificationBlock for usage in NoteTableViewCell, in the subject field
+    *
+    *	@returns	A Subject Attributed String
+    */
+    public func attributedSubjectText() -> NSAttributedString {
```
Comment (diegoreymendez) [inline WordPress/Classes/Extensions/Notifications/NotificationBlock+Interface.swift]: @jleandroperez - I'd consider unit-testing this public method.

File: WordPress/Classes/Extensions/Notifications/NotificationBlock+Interface.swift [hunk 3]
```diff
@@ -1,55 +1,120 @@
...
 
-    public func snippetAttributedText() -> NSAttributedString {
-        if text == nil {
-            return NSAttributedString()
+    /**
+    *	@brief      Formats a NotificationBlock for usage in NoteTableViewCell, in the snippet field
+    *
+    *	@returns	A Snippet Attributed String
+    */
+    public func attributedSnippetText() -> NSAttributedString {
```
Comment (diegoreymendez) [inline WordPress/Classes/Extensions/Notifications/NotificationBlock+Interface.swift]: @jleandroperez - I'd consider unit-testing this public method.

File: WordPress/Classes/Extensions/Notifications/NotificationBlock+Interface.swift [hunk 4]
```diff
@@ -1,55 +1,120 @@
...
 
-        let richSnippetCacheKey = "richSnippet"
-        if let cachedSnippet = cacheValueForKey(richSnippetCacheKey) as? NSAttributedString {
-            return cachedSnippet
+    /**
+    *	@brief      Formats a NotificationBlock for usage in NoteBlockHeaderTableViewCell
+    *
+    *	@returns	A Header Attributed String
+    */
+    public func attributedHeaderTitleText() -> NSAttributedString {
```
Comment (diegoreymendez) [inline WordPress/Classes/Extensions/Notifications/NotificationBlock+Interface.swift]: @jleandroperez - I'd consider unit-testing this public method.

**Comment by diegoreymendez** [comment]
@jleandroperez - I'd consider unit-testing the public methods in this class.

**Comment by jleandroperez** [comment]
@diegoreymendez if it's okay with you, let's address Notifications-Model + Helpers Unit Testing in a different issue: #3390

Thanks for the suggestion, i believe Unit Testing just needs some extra time to get used to, but it'll lead to better and cleaner API's. :heart:

**Comment by diegoreymendez** [comment]
:+1: - :shipit: (100% emoticon communication)

**Comment by jleandroperez** [comment]
thanks Diego!

**Pull request merged**